### PR TITLE
feat: chimpunk

### PR DESCRIPTION
**Pull request by GautierArcin**
I'm trying to implement newsletter 

I'm not sure about `mailchump marketing` in `package.json`, should they be in `dev dependencies` ?

Also, what do you think of it right now ?

(I'll try to add Buttondown later this week)

### CODE REVIEW

File: data/siteMetadata.js [hunk 1]
```diff
@@ -22,6 +22,15 @@ const siteMetadata = {
...
     googleAnalyticsId: '', // e.g. UA-000000-2 or G-XXXXXXX
   },
+  newsletter: {
+    provider: 'buttondown', // supported providers: mailchimp
+    mailChimpConfig: {
+      apiKey: process.env.NEXT_PUBLIC_MAILCHIMP_API_KEY,
+      apiServer: process.env.NEXT_PUBLIC_MAILCHIMP_API_SERVER,
+      audienceId: process.env.NEXT_PUBLIC_MAILCHIMP_AUDIENCE_ID,
+    },
+    buttonDown: process.env.NEXT_PUBLIC_BUTTONDOWN,
```
Comment (timlrx) [inline data/siteMetadata.js:32]: How about `buttondownConfig` for consistency? And we can change `disqus` below to `disqusConfig` as well

File: package.json [hunk 2]
```diff
@@ -12,12 +12,14 @@
...
   },
   "dependencies": {
+    "@mailchimp/mailchimp_marketing": "^3.0.58",
     "@tailwindcss/forms": "^0.3.2",
     "@tailwindcss/typography": "^0.4.0",
     "autoprefixer": "^10.2.5",
     "esbuild": "^0.12.15",
     "gray-matter": "^4.0.2",
     "image-size": "1.0.0",
+    "mailchimp": "^1.2.1",
```
Comment (timlrx) [inline package.json:22]: Looks like this is unused?

**Comment by timlrx** [comment]
Thanks for the feature, much appreciated! Not sure, where's the best place to add it or whether we should even insert it by default but I will take a look when you are done.

**Comment by GautierArcin** [comment]
Hey again ! Sorry for the delay, I'm currently passing interviews for new jobs.

@timlrx  I applied the point you asked me to correct, everything should be fine. What do you think of the current version ?

@DLevai94  Hello, I'm super sorry to tag you.

I've seen that you also implemented a newsletter on your blog. Would it be possible to take a look at your implementation, or to take a look at the implementation ?

Again, apologies for the unprompted tagging.

File: pages/api/buttondown.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// eslint-disable-next-line import/no-anonymous-default-export
+export default async (req, res) => {
+  const { email } = req.body
+
+  console.log('hello 1')
```
Comment (DLevai94) [inline pages/api/buttondown.js:5]: Unnecessary console.log

File: pages/api/buttondown.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  if (!email) {
+    return res.status(400).json({ error: 'Email is required' })
+  }
+
+  try {
+    const API_KEY = process.env.NEXT_PUBLIC_BUTTONDOWN
+    const response = await fetch(`https://api.buttondown.email/v1/subscribers`, {
```
Comment (DLevai94) [inline pages/api/buttondown.js:12]: It'd be great if the API route would come from either `.env` or a `const`

File: .env.example [hunk 5]
```diff
@@ -3,4 +3,8 @@ NEXT_PUBLIC_GISCUS_REPOSITORY_ID=
 NEXT_PUBLIC_GISCUS_CATEGORY=
 NEXT_PUBLIC_GISCUS_CATEGORY_ID=
 NEXT_PUBLIC_UTTERANCES_REPO=
-NEXT_PUBLIC_DISQUS_SHORTNAME=
\ No newline at end of file
+NEXT_PUBLIC_DISQUS_SHORTNAME=
+NEXT_PUBLIC_MAILCHIMP_API_KEY=
+NEXT_PUBLIC_MAILCHIMP_API_SERVER=
+NEXT_PUBLIC_MAILCHIMP_AUDIENCE_ID=
+NEXT_PUBLIC_BUTTONDOWN=
```
Comment (DLevai94) [inline .env.example:10]: The variables only used in API routes don't have to be prefixed with NEXT_PUBLIC. This way they're exposed to the browser too. https://nextjs.org/docs/basic-features/environment-variables

**Comment by DLevai94** [comment]
Hi @GautierArcin ,

Thank you for tagging me. I took a look at your PR and added a couple of comments.

Sure you can take a look at my implementation too. Here's the repo: https://github.com/DLevai94/garden
The files you need:
- `.env.example`
- `components/EmailCTA.js`
- `pages/api/subscribe.js`

Feel free to implement [ConvertKit](https://convertkit.com/) integration too based on my repo. And let me know if you have any questions.

**Comment by GautierArcin** [comment]
Hey @DLevai94,

Thank you a lot for your review. I applied all the changes you mentioned.

I began the implementation of Convertkit. I need to test it a bit more tomorrow, but it should be working.
Branch just need a little more of polish on the aesthetics side (I'm still not satisfied with button's design) and it should be ready to be merged !

Again, thanks a lot for your help (and your authorization for ConvertKit implementation) ! :smile:

File: pages/api/convertkit.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+  if (!list) {
+    return res.status(400).json({ error: 'List is required' })
+  }
+
+  try {
+    const FORM_ID =
+      list === '12-tips-production-apps'
+        ? process.env.CONVERTKIT_TIPS_FORM_ID
+        : process.env.CONVERTKIT_MONTHLY_FORM_ID
```
Comment (DLevai94) [inline pages/api/convertkit.js:17]: Only need to load 1 form ID here

File: pages/api/convertkit.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable import/no-anonymous-default-export */
+export default async (req, res) => {
+  const { email, list } = req.query
+
+  if (!email) {
+    return res.status(400).json({ error: 'Email is required' })
+  }
+
+  if (!list) {
```
Comment (DLevai94) [inline pages/api/convertkit.js:9]: Don't need list param when using 1 form

**Comment by DLevai94** [comment]
Nice work @GautierArcin . Added a few comments. I have 2 forms on my website, but in this starter, only 1 needed.

**Comment by GautierArcin** [comment]
Hey @DLevai94, thanks a lot for the feedback ! I changed the two form settings. It should be working now (I tested it with my own convertkit account).

@timlrx  I think the changes should be ready to merge with the master, apart from possible minor aesthetics change. What do you think ?

File: components/FormSuscribe.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import { useRef, useState } from 'react'
+
+import siteMetadata from '@/data/siteMetadata'
+
+const FormSuscribe = () => {
```
[the remark this answers is not in the document]
Comment (timlrx) [reply components/FormSuscribe.js]: Thanks for fixing the rest, but the component name is still spelled as FormSuscribe :)

**Comment by timlrx** [comment]
Hi @GautierArcin, I registered for Buttondown so that we can use it as a demo on the website. Have also modified the config.

We should fix the `autofill` style for the input. This is how dark mode looks like:
image

The position of the component is a little weird. Try moving it either to the right of "Latest" or maybe to the bottom of the page?

**Comment by GautierArcin** [comment]
@timlrx  The `FormSubscribe` typo should be fixed.

Also, I wasn't able to replicate the bug on the autofill for darkmode... I'm using firefox. It displays as that:

image

Is the style better like that ? (Could you take a particular look at `item-center` ? I'm not sure how to center it on the secondary axis of the flex).

**Comment by timlrx** [comment]
Thanks! Let me play around with the styles before merging it to master. I am using chrome so there might be a bit of difference.

**Pull request merged**